src/web_socket.ts

```diff
@@ -200,28 +200,37 @@ class Parser extends EventEmitter {
       return
     }
 
+    let event = ''
+    let payload = ''
+    let mes = {}
     try {
       const obj = JSON.parse(data)
-      const payload: string = obj.payload
-      const mes = JSON.parse(payload)
-      switch (obj.event) {
-        case 'update':
-          this.emit('update', mes as Status)
-          break
-        case 'notification':
-          this.emit('notification', mes as Notification)
-          break
-        case 'delete':
-          this.emit('delete', mes as number)
-          break
-        case 'conversation':
-          this.emit('conversation', mes as Conversation)
-          break
-        default:
-          this.emit('error', new Error(`Unknown event has received: ${obj}`))
-      }
+      event = obj.event
+      payload = obj.payload
+      mes = JSON.parse(payload)
     } catch (err) {
-      this.emit('error', new Error(`Error parsing websocket reply: ${data}, error message: ${err}`))
+      // delete event does not have json object
+      if (event !== 'delete') {
+        this.emit('error', new Error(`Error parsing websocket reply: ${data}, error message: ${err}`))
+        return
+      }
+    }
+
+    switch (event) {
+      case 'update':
+        this.emit('update', mes as Status)
+        break
+      case 'notification':
+        this.emit('notification', mes as Notification)
+        break
+      case 'conversation':
+        this.emit('conversation', mes as Conversation)
+        break
+      case 'delete':
+        this.emit('delete', payload as string)
```

Comment:
__[TSLint]__ This assertion is unnecessary since it does not change the type of the expression. (no-unnecessary-type-assertion) ([link](https://sider.review/gh/repos/136549493/pulls/42#issue-4117605))

<sub>posted by Sider</sub>